### PR TITLE
HDDS-4614. Add Integration test for HDDS upgrade: Happy path cases.

### PR DESCRIPTION
**Pull request by prashantpogde**
## What changes were proposed in this pull request?

Added integration test for SCM + Datanodes going through upgrades.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4614

## How was this patch tested?

Running the tests.

### CODE REVIEW

**Comment by prashantpogde** [comment]
CI failures are unrelated with the changes.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/upgrade/DataNodeUpgradeFinalizer.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.concurrent.Callable;
+
+import org.apache.hadoop.hdds.upgrade.HDDSLayoutFeatureCatalog.HDDSLayoutFeature;
+import org.apache.hadoop.ozone.container.common.statemachine.DatanodeStateMachine;
+import org.apache.hadoop.ozone.upgrade.BasicUpgradeFinalizer;
+
+/**
+ * UpgradeFinalizer for the DataNode.
+ */
+public class DataNodeUpgradeFinalizer extends
```
Comment (avijayanhwx) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/upgrade/DataNodeUpgradeFinalizer.java:35]: I believe _HDDSLayoutFeatureCatalog.HDDSLayoutFeature#FIRST_UPGRADE_VERSION_ is no longer needed.

**Comment by vivekratnavel** [comment]
@prashantpogde Can you please fix the checkstyle issues:

```
hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/upgrade/DataNodeUpgradeFinalizer.java
 39: &apos;,&apos; is preceded with whitespace.
 ```

**Comment by prashantpogde** [comment]
Rebased the changes and uploaded again.

**Review by avijayanhwx — APPROVED** [review]
LGTM +1.

**Pull request merged**